api/cds.proto

```diff
@@ -417,6 +415,15 @@ message Cluster {
   // For instance, if the metadata is intended for the Router filter, the filter
   // name should be specified as *envoy.router*.
   Metadata metadata = 25;
+
+  enum ClusterProtocolSelection {
+    // Cluster can only operate on one of the possible upstream protocols (HTTP1.1, HTTP2).
+    // If http2_protocol_options are present, HTTP2 will be used, otherwise HTTP1.1 will be used.
+    EXCLUSIVE_AS_CONFIGURED = 0;
+    // Use HTTP1.1 or HTTP2, depending on which one is used on the downstream connection.
+    USE_DOWNSTREAM_PROTO = 1;
```

Comment:
Prefer `USE_DOWNSTREAM_PROTOCOL` to avoid confusion with protobufs (I read it as "use the protobuf supplied from downstream to specify which protocol to use" at first glance)

Reply:
Good point, done :-)